### PR TITLE
Fix project module bugs

### PR DESCRIPTION
**Pull request by GauravGusain98**
Targets #1352

### Description

- Added null checks for the project contract functionality
- Added null check for project contract while creating the project
- Added search functionality in project listing
- Sort project listing by name
- Make Contract file field required

### Checklist:

- [x] I have performed a self-review of my own code.

### CODE REVIEW

File: Modules/Project/Resources/views/index.blade.php [hunk 1]
```diff
@@ -2,30 +2,43 @@
...
-    <div class="d-flex justify-content-between mb-2 align-items-center">
-       <div class="d-md-flex">
-            <h4 class="">{{ config('project.status')[request()->input('status', 'active')] }} Projects
-                ({{ $projects ? $projects->count() : ''}})</h4>
-            <form class="mx-md-3" action="{{ route('project.index', ['status' => 'active'])  }}">
+    <div class='d-none d-md-flex justify-content-between'>
+        @include('project::menu_header')
+        @can('projects.create')
+            <span class='mt-4'>
+                <a href= "{{ route('project.create') }}" class="btn btn-info text-white"> Add new project</a>
```
Comment (kuldeep3) [inline Modules/Project/Resources/views/index.blade.php:9]:
```suggestion
                <a href= "{{ route('project.create') }}" class="btn btn-info text-white">Add new project</a>
```

File: Modules/Project/Resources/views/index.blade.php [hunk 2]
```diff
@@ -2,30 +2,43 @@
...
+            </div>
+        </form>
+    </div>
+    <div class='d-md-none mb-2'>
         @can('projects.create')
-        <span>
-            <a  href= "{{ route('project.create') }}" class="btn btn-info text-white"> Add new project</a>
-        </span>
+            <div class="d-flex flex-row-reverse">
+                <a href= "{{ route('project.create') }}" class="btn btn-info text-white"> Add new project</a>
```
Comment (kuldeep3) [inline Modules/Project/Resources/views/index.blade.php:37]:
```suggestion
                <a href= "{{ route('project.create') }}" class="btn btn-info text-white">Add new project</a>
```

**Review by kuldeep3 — APPROVED** [review]
LGTM 👍🏽  Two minor changes.

File: Modules/Project/Resources/views/index.blade.php (hunk 2 above)
Comment (GauravGusain98) [reply Modules/Project/Resources/views/index.blade.php]: Done

**Pull request merged**